### ISSUE TITLE
Linking Sign in and Sign Up Flows

### ISSUE THREAD

**Issue by dmtrjsg** [Joystream/atlas#4335]
## Context
We discovered there's no support for the user with existing Orion account (if they are already logged in) to sign up to YPP without being forced to create a brand new acc.

## Scope
- [ ] Interlink sign in and sign up flows to allow user with existing account to sign up to YPP in a single flow

**Comment by KubaMikolajczyk** [comment]
I prepared a solution and proposed it in this short video update along with my thought process on that ✅
🎥 Loom video: https://www.loom.com/share/868c5a661812410da6c24c5ba72feda3?sid=dc1f1a50-a265-4ae5-a435-6d5b9d4c6d89
📄 Exploration files: https://www.figma.com/file/ymHYOFItrmFaLeBC5Z4tpQ/Sign-up-%2F-Log-in?type=design&node-id=3204-77453&mode=design&t=T0zZ1vmCGmtWbIky-4

**Comment by dmtrjsg** [comment]
@KubaMikolajczyk just one more small thing left [here> ](https://www.figma.com/file/ymHYOFItrmFaLeBC5Z4tpQ/Sign-up-%2F-Log-in?type=design&mode=design&t=PjqD72URIPRk0BqT-0#481416568)

**Comment by KubaMikolajczyk** [comment]
@dmtrjsg @attemka @drillprop I finalized the linking of sign in and create account experiences but more changes followed the issues which was uncovered during our discussion with Bartosz. So in this video update I will cover not only the linking of those experiences, but also the updated flows for creating account, changes in dialogs (titles, button labels etc) and changes in YPP sign up experience - as always please share your feedback here ✅
🎥 Loom video: https://www.loom.com/share/63a7709b8eac4389bf879c9bf7f7fe86?sid=c9731395-5bc9-49dd-8564-f53a0a97b153
📄 Sign in / Create account changelog: https://www.figma.com/file/ymHYOFItrmFaLeBC5Z4tpQ/Sign-up-%2F-Log-in?type=design&node-id=2013-33819&mode=design&t=iHdZ7IfrZYmaiNWQ-4
📄 Video changelog: https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video?type=design&node-id=6396-297910&mode=design&t=Av7kqs1U3Yf0zZ6c-4
📄 Channel changelog: https://www.figma.com/file/POw4j95jszfIP5bGfas2IC/Channel?type=design&node-id=2386-171791&mode=design&t=bvAIsin7N944oEby-4
📄YPP Changelog: https://www.figma.com/file/oQqFqdAiPu16eeE2aA5AD5/YouTube-Partner-Program?type=design&node-id=3338-84773&mode=design&t=EdWK52Gag8lxK17Q-4

**Comment by dmtrjsg** [comment]
1️⃣ I like the idea of asking if they want to create account or sign in on the requirements page, seem to be solving a lot of problems. 

2️⃣ I don't think this is a good idea as it adds a step for most of the users and complicate the already complex feature, while catering to a _very low %_ of beneficiaries: 
<img width="258" alt="Screenshot 2023-06-27 at 16 10 17" src="https://github.com/Joystream/atlas/assets/93924525/b14c64ba-091b-4b3b-b3e4-ad797c566b86">

3️⃣  I believe that we already implemented the email to be shown as the first step after the Google Auth page, would that be okay to keep this order? (Remember when we agreed that returning user to the email page in case of email is duplicate is okay since this will never happen in practice?)

**Comment by KubaMikolajczyk** [comment]
Sorry for making it hard to follow the whole issue was harder then expected :)
1. Great!
2. This step is already in the flow and was there before - its necessary to allow user to choose the channel they want to go with. The only change I'm proposing here is to add create new channel button which will allow user to do exactly that. So we wont force them to use the channels that they dont want to use (and they won't be stuck if using one of those that they own already was not on their mind)
3. The current implementation in YPP is -> seed ->email -> password - Bartosz didn't know that at the time when you spoke with him. And yes we agreed on email - password -seed order because previously we thought that it is not possible to show seed first as it was supposed to be generated from email and password - and in that case the order didn't matter as we couldn't achieve the best scenario anyway. But If the seed is shown first then I would vote strongly for keeping email last to accomodate for those error scenarios of already used email.

**Comment by dmtrjsg** [comment]
2. Ok let's do it, 3. also fine 👍

**Comment by dmtrjsg** [comment]
According to Kuba, this whole changelog of changes was not merged:
[Link to figma](https://www.figma.com/file/ymHYOFItrmFaLeBC5Z4tpQ/Sign-up-%2F-Log-in?type=design&node-id=1284-11682&mode=design&t=GbdtJVwrNLblVF5G-0
)

<img width="676" alt="Screenshot 2023-07-06 at 15 50 48" src="https://github.com/Joystream/atlas/assets/93924525/ef795b2d-2e01-4616-881b-e2bfda7d6f5e">
